src/sentry/event_manager.py

```diff
@@ -1878,7 +1878,10 @@ def _create_group(project: Project, event: Event, **kwargs: Any) -> Group:
     group_data["metadata"].update(severity)
 
     if features.has("projects:issue-priority", project, actor=None):
-        priority = _get_priority_for_group(severity, kwargs)
+        priority = kwargs.get("priority", None)
```

Comment:
Ensures we don't override the priority with log level logic if it's already provided